app/utils/vcs.js

```diff
@@ -0,0 +1,59 @@
+import config from 'travis/config/environment';
+
+const { providers } = config;
+
+const vcsId = (vcsType) => {
+  const availableProvidersRegex = Object.keys(providers).join('|');
+  const match = (vcsType || 'github').match(new RegExp(availableProvidersRegex, 'i'));
+
+  if (!match[0]) {
+    throw new Error(`Invalid VCS Type "${vcsType}"`);
+  }
+  return match[0].toLowerCase();
+};
+
+const vcsConfig = (vcsType) => (
+  providers[vcsId(vcsType)]
+);
+
+const replaceParams = (template, params) => (
+  Object
+    .keys(params)
+    .reduce((url, key) => url.replace(`:${key}`, params[key]), template)
+);
+
+const templateParams = (template) => (
+  template
+    .match(/:[a-z]+/g)
+    .map((param) => param.slice(1))
+);
+
+const paramsValid = (template, params) => (
+  arrayEqual(Object.keys(params), templateParams(template))
+);
+
+const arrayEqual = (array1, array2) => (
+  array1.sort().toString() == array2.sort().toString()
+);
+
+export const vcsUrl = (resource, vcsType, params) => {
+  const vcs = vcsConfig(vcsType);
+  const url = vcs.endpoint + vcs.paths[resource];
+
+  if (!paramsValid(url, params)) {
+    throw new Error(`Missing url params. URL: ${url}, PARAMS: ${JSON.stringify(params)}`);
```

Comment:
You can use [`assert`](https://api.emberjs.com/ember/3.12/functions/@ember%2Fdebug/assert) method in such cases. Ember CLI removes them from production builds.